lib/tasks/pkg.rake

```diff
@@ -23,6 +23,9 @@ namespace :pkg do
     ref = ENV['ref'] || 'HEAD'
     version = `git show #{ref}:VERSION`.chomp.chomp('-develop')
     raise "can't find VERSION from #{ref}" if version.length == 0
-    `git archive --prefix=foreman-#{version}/ #{ref} | bzip2 -9 > pkg/foreman-#{version}.tar.bz2`
+    `git archive --prefix=foreman-#{version}/ #{ref} > pkg/foreman-#{version}.tar`
+    `npm install`
+    `tar --transform 's,^,foreman-#{version}/,' -r -f pkg/foreman-#{version}.tar node_modules`
```

Comment:
Notice how Cockpit (packaged for Fedora & others) runs `npm install` to generate the tarball here:

https://github.com/cockpit-project/cockpit/blob/a0a7d1e1a0589a77ae4c068a1c215bb67c31448d/autogen.sh
https://github.com/cockpit-project/cockpit/blob/8935481284d9f70b56f42ce063eb2ad7af123b39/tools/make-source

Then the spec uses said tarball or an 'official release' tarball (https://github.com/cockpit-project/cockpit/releases/) as Source0 in the spec. Both contain the node_modules, minus whatever modules are used only in development or testing.
https://github.com/cockpit-project/cockpit/blob/master/tools/cockpit.spec